diff -urp src/server/shared/Cryptography/OpenSSLCrypto.cpp src/server/shared/Cryptography/OpenSSLCrypto.cpp
--- src/server/shared/Cryptography/OpenSSLCrypto.cpp	2013-10-27 16:44:23.000000000 +0100
+++ src/server/shared/Cryptography/OpenSSLCrypto.cpp	2013-10-27 16:45:13.000000000 +0100
@@ -33,27 +33,27 @@ static void lockingCallback(int mode, in
 
 static void threadIdCallback(CRYPTO_THREADID * id)
 {
-    CRYPTO_THREADID_set_numeric(id, ACE_Thread::self());
+//    CRYPTO_THREADID_set_numeric(id, ACE_Thread::self());
 }
 
 void OpenSSLCrypto::threadsSetup()
 {
-    cryptoLocks.resize(CRYPTO_num_locks());
-    for(int i = 0 ; i < CRYPTO_num_locks(); ++i)
-    {
-        cryptoLocks[i] = new ACE_Thread_Mutex();
-    }
-    CRYPTO_THREADID_set_callback(threadIdCallback);
-    CRYPTO_set_locking_callback(lockingCallback);
+//    cryptoLocks.resize(CRYPTO_num_locks());
+//    for(int i = 0 ; i < CRYPTO_num_locks(); ++i)
+//    {
+//        cryptoLocks[i] = new ACE_Thread_Mutex();
+//    }
+//    CRYPTO_THREADID_set_callback(threadIdCallback);
+//    CRYPTO_set_locking_callback(lockingCallback);
 }
 
 void OpenSSLCrypto::threadsCleanup()
 {
-    CRYPTO_set_locking_callback(NULL);
-    CRYPTO_THREADID_set_callback(NULL);
-    for(int i = 0 ; i < CRYPTO_num_locks(); ++i)
-    {
-        delete cryptoLocks[i];
-    }
-    cryptoLocks.resize(0);
+//    CRYPTO_set_locking_callback(NULL);
+//    CRYPTO_THREADID_set_callback(NULL);
+//    for(int i = 0 ; i < CRYPTO_num_locks(); ++i)
+//    {
+//        delete cryptoLocks[i];
+//    }
+//    cryptoLocks.resize(0);
 }
\ No newline at end of file
